From ecfd78d190fb224ee86ae9502f7913565eb579af Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Fri, 5 Sep 2014 11:03:48 -0400
Subject: [ethernet] sfc: Preserve rx_frm_trunc counters when resizing DMA rings

Message-id: <1409915048-26438-13-git-send-email-nikolay@redhat.com>
Patchwork-id: 91409
O-Subject: [RHEL7.1 PATCH BZ 1110888 12/32] sfc: Preserve rx_frm_trunc counters when resizing DMA rings
Bugzilla: 1110888
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Ben Hutchings <bhutchings@solarflare.com>

We allocate efx_channel structures with kzalloc() so we don't need to
zero-initialise individual fields in efx_probe_channel().  Further,
this function will be called again during DMA ring resizing and we
should not reset any statistics then.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: Shradha Shah <sshah@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 2fa25cf1e1cad74df4b465710e55a62386743614)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/sfc/efx.c b/drivers/net/ethernet/sfc/efx.c
index d16c94c..6f7a86f 100644
--- a/drivers/net/ethernet/sfc/efx.c
+++ b/drivers/net/ethernet/sfc/efx.c
@@ -503,8 +503,6 @@ static int efx_probe_channel(struct efx_channel *channel)
    goto fail;
  }
 
- channel->n_rx_frm_trunc = 0;
-
  return 0;
 
 fail:
-- 
1.7.1